tarpc/examples/custom_transport.rs

```diff
@@ -0,0 +1,60 @@
+use futures::future;
+use tarpc::context::Context;
+use tarpc::serde_transport as transport;
+use tarpc::server::{BaseChannel, Channel};
+use tokio::net::{UnixListener, UnixStream};
+use tokio_serde::formats::Bincode;
+use tokio_util::codec::{length_delimited::LengthDelimitedCodec, Framed};
+
+#[tarpc::service]
+pub trait PingService {
+    async fn ping();
+}
+
+#[derive(Clone)]
+struct Service;
+
+impl PingService for Service {
+    type PingFut = future::Ready<()>;
+
+    fn ping(self, _: Context) -> Self::PingFut {
+        future::ready(())
+    }
+}
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
+    let bind_addr = "/tmp/tarpc_on_unix_example.sock";
+
+    let _ = std::fs::remove_file(bind_addr);
+
+    let (tx_started, rx_started) = tokio::sync::oneshot::channel();
+
+    tokio::spawn(async move {
+        let listener = UnixListener::bind(bind_addr).unwrap();
+        let codec_builder = LengthDelimitedCodec::builder();
```

Comment:
nit: if you do these 2 lines above the spawned async block, then you don't need to signal via channel when the UnixListener is ready.